### ISSUE TITLE
Mentions sur FACTURE - modifications URGENTES

### ISSUE THREAD

**Issue by lgazeau** [lgi-wines/Extranet-LGI#150]
Suite contrôle, nous devons mettre à jour nos MENTIONS SUR FACTURE
 par rapport au Tableau qui est en PJ

Merci de m'appeler si c'est confus, car pas simple

[Mentions sur FACTURE.xlsx](https://github.com/lgi-wines/Extranet-LGI/files/1667935/Mentions.sur.FACTURE.xlsx)

**Comment by Portekoi** [comment]
Je n'ai pas tout compris. Il me faudrait des exemples de `Kart` pour chaque modification avec au stylo par exemple ce qu'il faut ajouter. Merci

**Comment by lgazeau** [comment]
En PJ, des exemples de **KartID** pour chaque cas de figure.

[Uploading Mentions sur FACTURE v2.xlsx…]()

A ta disposition.

**Comment by Portekoi** [comment]
Ca n'a pas fonctionné je pense. Le lien ne donne pas de fichier.

**Comment by lgazeau** [comment]
[Mentions sur FACTURE v2.xlsx](https://github.com/lgi-wines/Extranet-LGI/files/1684996/Mentions.sur.FACTURE.v2.xlsx)

**Comment by Portekoi** [comment]
Je pige pas : je dois ajouter ce qui est en rouge? Il faut qu'on s'appelle je pense pour que je comprenne bien.

**Comment by lgazeau** [comment]
Tu dois effectivement ajouter ce qui est en rouge.
Il faut "au préalable" créer dans le Formulaire Client le champ "N° Autorisation Achats Franchise TVA" (on parle de celui du client)
L'ADV alimentera ce champ => on le récupérera au niveau de la Facture, derrière la mention "Art. N° 275" pour Hors UE d'un client FRANCE.
Suis dispo au tel à partir de 13h30 au 04 68 11 60 80 // 06 11 17 88 83 ou dès maintenant au besoin

**Comment by Portekoi** [comment]
Ok je vois ça mardi. Je t'appelle si j'ai besoin.

**Comment by Portekoi** [comment]
Pour le cas `pour Hors UE d'un client FRANCE`, quelle est la phrase complète à ajouter? 

`Goods sold without VAT, in application of Art. N° 275 + N°  Autorisation Achats Franchise TVA from 'Code Général des Impôts'` en dessous?

**Comment by Portekoi** [comment]
Je viens d'ajouter le champ au niveau du client : 
![image](https://user-images.githubusercontent.com/1560754/37626648-2735759c-2bd1-11e8-976a-a169e2268af2.png)

```sql

ALTER TABLE pinard.dbo.Client
ADD AAFTVANumber varchar(50) NULL; 

```

**Comment by Portekoi** [comment]
@lgazeau Tu peux voir mon commentaire un peu plus haut ? merci

https://github.com/lgi-wines/Extranet-LGI/issues/150#issuecomment-374380784

**Comment by lgazeau** [comment]
Mention à mettre en français puisqu'on facture une entreprise française, et qu'il s'agit d'un document qui est appelé à être contrôlé en France :
"Marchandises vendues hors taxe, en vertu de l'Art. N° 275 du C.G.I.
Autorisation N° X" (X étant à récupérer dans le Formulaire Client, à partir du champ que tu as créé)

**Comment by lgazeau** [comment]
PARFAIT

**Comment by Portekoi** [comment]
Est-ce le résultat attendu pour le `Kart` `36583`?

[Kart_Invoice_36583_20180320_222816 (1).pdf](https://github.com/lgi-wines/Extranet-LGI/files/1831273/Kart_Invoice_36583_20180320_222816.1.pdf)

**Comment by Portekoi** [comment]
@lgazeau J'attends ton retour sur ce point.

**Comment by lgazeau** [comment]
OK
J'aurais juste mis la mention avant le pavé lié aux aspects douaniers (sous le signature), au lieu de la mettre dans le trafic.

**Comment by lgazeau** [comment]
...avec le "N° Autorisation Achats Franchise" à récupérer dans le formulaire Client bien évidemment.

**Comment by Portekoi** [comment]
OK je ferai la modification.

**Comment by lgazeau** [comment]
OK on attend

**Comment by Portekoi** [comment]
Pour validation. Cela ne s'appliquera que pour les clients facturés en France mais donc le pays de livraison est hors d'Europe : 

![image](https://user-images.githubusercontent.com/1560754/37702013-77bb5e3e-2cf1-11e8-9550-3a093d26a787.png)

**Comment by lgazeau** [comment]
Oui,
Et pour les Facturations NÉGOCE des clients "Hors UE" il faut remplacer :
"Art. N° 262" par "Art. N° 262-1" (rajout de l'alinéa "1")
=> voir PJ
[Mentions sur FACTURE v2.xlsx](https://github.com/lgi-wines/Extranet-LGI/files/1832613/Mentions.sur.FACTURE.v2.xlsx)

**Comment by Portekoi** [comment]
Si je prends le `Kart` en exemple n° `37467`, j'ai déjà un `I` : 

![image](https://user-images.githubusercontent.com/1560754/37702728-c21b6c6a-2cf3-11e8-8733-28d28eadde1a.png)

Je dois remplacer la lettre `I` par le chiffre `1` ?

**Comment by lgazeau** [comment]
La modification avait donc été faite, mais il faut bien remplacer la lettre I par le chiffre 1 pour ves ventes Hors UE uniquement.

**Comment by Portekoi** [comment]
Ok j'ai bien effectué les modifications mais je livrerai ce soir pour ne pas vous gêner en cas de soucis.

**Comment by lgazeau** [comment]
OK

**Comment by Portekoi** [comment]
Exemple en ligne : http://lgivm.waybis.com/uploads/Karts/Kart_Invoice_36583_20180321_223141.pdf

A noter que j'ai ajouté un `N° Autorisation Achats Franchise` au client `VINADEIS DISTRIBUTION` pour que la mention apparaisse. 

*Si  `N° Autorisation Achats Franchise` est vide, **la mention n'apparaitra pas**.*

C'est bon pour le remplacement du `I` par le `1` dans `n°262-I`

**Comment by Portekoi** [comment]
J'ai publié hier soir. Donc c'est en ligne.

**Comment by lgazeau** [comment]
Cela n'est pas opérationnel.
Ni sur "Editer Facture Négoce" ni sur "Facture PDF"
=> voir PJ
[Invoice KartID 36583.pdf](https://github.com/lgi-wines/Extranet-LGI/files/1836506/Invoice.KartID.36583.pdf)

**Comment by Portekoi** [comment]
Est ce que le numéro d'autorisation a bien été saisie pour ce client?

**Comment by Portekoi** [comment]
Je viens de voir que le numéro d'autorisation n'a pas été saisie

**Comment by lgazeau** [comment]
C'est exact Monseigneur.
Il faut cependant afficher la mention (dans ce cas de figure bien évidemment), même si le numéro d'autorisation n'a pas été saisi.
A nous de le voir, et d'aller le compléter.
Quoi qu'il en soit, on (Alexandra ADV) va s'atteler à les renseigner.

**Comment by Portekoi** [comment]
Ok je relivre ce midi

**Comment by lgazeau** [comment]
Je pense que tu n'as pas eu le temps d'implémenter car j'ai supprimé le N° d'Autorisation de VINADEIS
ClientID=1001
et le Kart 36853 ne remonte pas la mention.
Tu me diras.

**Comment by Portekoi** [comment]
Effectivement, j'ai dû régler une autre urgence. Je livre ce soir sans faute.

**Comment by lgazeau** [comment]
OK

**Comment by Portekoi** [comment]
C'est en ligne : http://lgivm.waybis.com/uploads/Karts/Kart_Invoice_36583_20180322_213210.pdf

**Comment by Portekoi** [comment]
Est-ce que tu as pu tester?

**Comment by lgazeau** [comment]
Cela me semble OK.

**Comment by Portekoi** [comment]
Ok. On surveille et je te laisse clôturer si tout est ok pour vous.
